examples/nomad-consul-separate-cluster/main.tf

```diff
@@ -173,7 +173,6 @@ module "nomad_clients" {
   cluster_name  = "${var.nomad_cluster_name}-client"
   instance_type = "t2.micro"
 
-  # Give the clients a different tag so they don't try to join the server cluster
```

Comment:
Why is this being removed? Please note there are consul servers in this cluster that auto-discover each other using tags, so it's important that these other servers don't use the same tags as those consul servers.

Reply:
This the information I perceived to be misleading. Myself and several other folks I work with interpreted "client" and "server" in this sentence to mean "Nomad client" and "Nomad server", because this is the `nomad_clients` resource. With that misunderstanding, I thought that this example was attempting to configure Nomad to use the [cloud auto-join](https://learn.hashicorp.com/tutorials/nomad/clustering#join-nodes-using-cloud-auto-join) strategy to form a Nomad cluster, but really it uses the [Consul strategy](https://learn.hashicorp.com/tutorials/nomad/clustering#use-consul-to-automatically-cluster-nodes) to form clusters.

I think you're saying that this is sentence is meant to be interpreted as

> Give the Nomad client instances a different cluster tag so that the Consul agents running in client mode on these instances don't try to join the Consul server cluster.

If that's the case, something still isn't connecting for me. Isn't it desirable for the Consul agent's running in client mode on the Nomad client instances to join the Consul server cluster? If not, what Consul server cluster _should_ they join? As far as I can tell, there is only one Consul server cluster in this example.

Reply:
No, what I'm saying is that the way the Consul servers know how many servers there are, and which nodes are servers, is using tags. Any node that isn't a Consul server shouldn't be using the same tags as the servers.

Reply:
I was under the impression that a Consul agent acts as a server if it is given the `-server` flag (vs the `-client` flag) . The [run-consul](https://github.com/hashicorp/terraform-aws-consul/blob/master/modules/run-consul/run-consul#L35) script requires one of these flags.

In this example,

* Consul servers use tag `consul-servers: consul-example`
* Nomad servers use tag `nomad-servers: auto-join`
* Nomad clients use tag `nomad-clients: nomad-example`

I think it's misleading to say "Give the clients a different tag so they don't try to join the server cluster", because they're already using a different tag than the Consul server cluster.

We _do_ want the Consul clients on the Nomad clients to join the Consul cluster (as clients). That's taken care of by a different part of this example: the tags sent to the `run-consul` script that match the tags on the Consul server ASG:

```
data "template_file" "user_data_nomad_client" {
  template = file("${path.module}/user-data-nomad-client.sh")

  vars = {
    cluster_tag_key   = var.cluster_tag_key
    cluster_tag_value = var.consul_cluster_name
  }
}
```

Reply:
I think the piece you're missing is that the `run-consul` code uses the tags to find the ASG it's in, and then the size of the ASG to set the `bootstrap_expect` param: https://github.com/hashicorp/terraform-aws-consul/blob/master/modules/run-consul/run-consul#L274-L277.